src/track/beats.cpp

```diff
@@ -37,20 +166,656 @@ double Beats::findNBeatsFromSample(double fromSample, double beats) const {
         nthBeat = findNthBeat(prevBeat, fullBeats - 1);
     }
 
-    if (nthBeat == -1) {
-        return fromSample;
+    if (nthBeat == FramePos(-1)) {
+        return fromFrame;
     }
 
     // Add the fraction of the beat
     if (fractionBeats != 0) {
         nextBeat = findNthBeat(nthBeat, 2);
-        if (nextBeat == -1) {
-            return fromSample;
+        if (nextBeat == FramePos(-1)) {
+            return fromFrame;
         }
         nthBeat += (nextBeat - nthBeat) * fractionBeats;
     }
 
     return nthBeat;
 };
 
+void Beats::onBeatlistChanged() {
```

Comment:
normally functions are called by what they are doing and not when they are called. 
How about "updateBpm()"

Reply:
Yeah. This sounds like JavaScript. Will change this.